e3db/tsv1_auth.py

```diff
@@ -0,0 +1,47 @@
+from e3db.tests.test_tsv1_auth import PUBLIC_KEY_BYTES
+from nacl import public
+import requests
+from requests import auth
+from requests.auth import AuthBase
+from .sodium_crypto import SodiumCrypto
+from .base_crypto import BaseCrypto
+import datetime 
+
+class E3DBTSV1Auth(AuthBase):
+    HASHING_ALGORITHM = "BLAKE2B"
+    SIGNATURE_TYPE = "ED25519"
+    AUTHENTICATION_METHOD = "TSV1-" + SIGNATURE_TYPE + "-" + HASHING_ALGORITHM
+    AUTHORIZATION_HEADER = "Authorization"
+    SECRET_KEY_BYTES = 32
+
+    def __init__(self, private_signing_key: str, client_id: str):
+        self.private_signing_key = private_signing_key
+        self.client_id = client_id
+
+    def __call__(self, r: requests.Request):
+        """
+        TSV1 Authentication
+        """
+        private_b64_decoded = BaseCrypto.base64decode(self.private_signing_key)[:self.SECRET_KEY_BYTES]
```

Comment:
do better than me for the go implementation and add some story comments in between these blocks of code, e.g.

Derive and decode the signing keys

calculate header values

hash it

sign it

add authorization headers